taqueria-vscode-extension-web-ui/src/App.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useEffect, useState } from 'react';
+import { AllEditors } from './AllEditors';
+import { MichelineEditor } from './data-editors/MichelineEditor';
+import { InteropMessageInterface } from './interopTypes';
+
+const state = {
+	interop: undefined as undefined | InteropMessageInterface,
+	onChangeCallback: (x: InteropMessageInterface) => {/* empty */},
+	logs: [] as { message: string; data?: unknown }[],
+};
+
+// const log = (message: string, data?: unknown) => {
+// 	state.logs.push({ message, data });
+// };
+// console.log = log;
+
+const setGlobalInteropMessageInterface = (x: InteropMessageInterface) => {
+	state.interop = x;
+	state.onChangeCallback(x);
+};
+const w = window as unknown as Record<string, unknown>;
+w.setGlobalInteropMessageInterface = setGlobalInteropMessageInterface;
+
+export const App = () => {
+	const [deps, setDeps] = useState(state.interop);
+	useEffect(() => {
+		state.onChangeCallback = x => {
+			setDeps(x);
+		};
+		return () => {
+			state.onChangeCallback = () => {};
+		};
+	}, []);
```

Comment:
Should the dependencies for the effect be, `[state, setDeps]`?

Reply:
@ricklove

Reply:
setter should be constant, React does not change it in `useState`. Now on first component mount we just assign the outside `onChangeCallback` to update app component state